### DISCUSSION TITLE
React Native Expo, SocketIO not working on Android Phones.

### DISCUSSION THREAD

**Discussion by aju-alen** [socketio/socket.io#4942] [Q&A]
SocketIO connections are established between ios to ios, ios to windows but not from andrioid to ios or android to windows.

Client Socket connection 
```
import { io } from "socket.io-client";
import {Platform} from 'react-native';

export const socket = io.connect(`http://${myIP}:3000`);
```
What am I doing wrong?

**Comment by darrachequesne** [comment]
Hi! I think you need to allow plaintext HTTP:

```xml
<!-- res/xml/network_security_config.xml -->
<?xml version="1.0" encoding="utf-8"?>
<network-security-config>
  <domain-config cleartextTrafficPermitted="true">
    <domain includeSubdomains="true">$yourIP</domain>
  </domain-config>
</network-security-config>
```

References:

- https://developer.android.com/privacy-and-security/risks/cleartext
- https://stackoverflow.com/questions/45940861/android-8-cleartext-http-traffic-not-permitted
